src/core/PlayArea.ttslua

```diff
@@ -54,26 +56,31 @@ end
 -- data to the local token manager instance.
 ---@param args Table Single-value array holding the GUID of the Custom Data Helper making the call
 function updateLocations(args)
-  local customDataHelper = getObjectFromGUID(args[1])
+  customDataHelper = getObjectFromGUID(args[1])
   if customDataHelper ~= nil then
     tokenManager.addLocationData(customDataHelper.getTable("LOCATIONS_DATA"))
   end
 end
 
 function onCollisionEnter(collision_info)
-  if not COLLISION_ENABLED then return end
+  local obj = collision_info.collision_object
+
+  -- only trigger the following once every 0.1s for each object
+  -- (otherwise TTS fires this event 20 times and that causes 20 JSON decodes)
+  if not COLLISION_ENABLED or lastCollisionObject == obj then return end
```

Comment:
The multiple call only seems to happen for tokens.  Let's limit this to cards only (which is all we really care about) rather than the time gate

Reply:
Changed